### PR TITLE
Add RLVE environment support & starter pack

### PR DESCRIPTION
**Pull request by jbarnes850**
Developers often need a broad, verifiable, and adaptive environment suite to scale RL training. [RLVE](https://github.com/Zhiyuan-Zeng/RLVE) ("Scaling Up Reinforcement Learning for Language Models with Adaptive Verifiable Environments") offers 400+ environments with deterministic verifiers and difficulty controllers.

This pull request introduces a comprehensive RLVE integration for slime, enabling on-the-fly prompt generation and deterministic reward computation using a curated starter pack of math and logic environments with a one-line install `pip install rlve-gym` ([PyPI](https://pypi.org/project/rlve-gym/)).

The integration supports curriculum learning via adaptive environment sampling and difficulty controllers, and includes config files, docs, core provider and reward logic, and scripts for training with instruct models.

>  To use all 400+ environments: Clone the full RLVE repo, add to PYTHONPATH, and add entries to the YAML config. The integration architecture supports this without modification.

## Implementation: 

> note: all changes are under examples/rlve/; slime core is untouched

Slime's rollout path requires `args.rollout_global_dataset == True` by default. Rather than modifying slime core, this integration uses a **dummy indices  dataset** (`data/dummy_indices.jsonl`) containing 10,000 `{"index": N}` entries to satisfy slime's global dataset requirement. The custom generate function then replaces prompts on-the-fly with fresh RLVE problems:

```
  slime rollout loop
        |
        v
  rlve_generate.py    # Ignores dummy index, generates fresh prompt from RLVE
        |
        v
  rlve_reward.py      # Restores env state from metadata, runs verifier
        |
        v
 reward-key accuracy   # Binary signal (0/1) for training
```

This approach enables on-the-fly generation from verifiable environments without core slime changes.

## Changes

  **RLVE Integration and Prompt Generation:**
  * Added `rlve_prompt_provider.py` - `RLVEPromptProvider` class that samples environments by weight, generates problems with `problem_generation_seed` tracking
  (matching RLVE's `RLVEManager`), and supports curriculum learning with difficulty controllers
  * Added `rlve_generate.py` - Custom generate function that creates fresh prompts per rollout via the provider singleton
  * Added `__init__.py` - Module API exposing provider and utility functions

  **Deterministic Reward Computation:**
  * Added `rlve_reward.py` - Custom reward function using RLVE environment verifiers, with support for format scoring and curriculum controller updates

  **Configuration and Data:**
  * Added `configs/starter_pack.yaml` - Configuration specifying 7 starter environments (Multiplication, Division, Sorting, EuclidGame, ShortestPath,
  SpiralMatrix, LightUpPuzzle), weights, difficulty parameters, and answer format markers
  * Added `data/dummy_indices.jsonl` - 10,000 index entries to satisfy slime's global dataset requirement while enabling on-the-fly prompt generation

  **Documentation:**
  * Added `README.md` - Installation, quick start, environment details, configuration, state persistence, and troubleshooting

  **Training Script:**
  * Added `run_qwen3_8B_instruct.sh` - Training script for Qwen3-8B-Instruct on 2xH100 with recommended settings

  ## Starter Pack Environments

  | Environment | Weight | Key Parameter |
  |-------------|--------|---------------|
  | Multiplication | 1.0 | `digit_num` |
  | Division | 1.0 | `divisor_digit_num`, `answer_digit_num` |
  | Sorting | 1.0 | `N` (array length) |
  | EuclidGame | 1.0 | `MAX_X_Y` |
  | ShortestPath | 1.0 | `N`, `edge_density` |
  | SpiralMatrix | 1.0 | `MAX_M_N` |
  | LightUpPuzzle | 1.0 | `MAX_N_M`, `density` |

  All have fast, deterministic verifiers suitable for true-on-policy training.

  ## Verified on 2xH100

  Successfully ran 10 rollouts with Qwen3-8B on 2xH100:
  - RLVE prompt provider initializes from config
  - All 7 environments are sampled and verified
  - Training steps complete with proper accuracy feedback
  - Job completes successfully

  ## Usage

  ```bash
  pip install rlve-gym
  export RLVE_CONFIG_PATH=/path/to/configs/starter_pack.yaml

  # Run training
  ./examples/rlve/run_qwen3_8B_instruct.sh

### CODE REVIEW

**Comment by zhaochenyang20** [comment]
Great work! Let Zhiyuan and Zilin take a look.

**Comment by zhuzilin** [comment]
Thank you for this PR. But I'm afraid I have to close as this doesn't seem to be a nice example for slime (which should be simple and be the guide for user to use some specific feature that the doc does not cover). It suits to be an isolated repo, e.g. [Alibaba-NLP/qqr](https://github.com/Alibaba-NLP/qqr).
